core/tx_pool.go

```diff
@@ -1098,6 +1101,11 @@ func (pool *TxPool) scheduleReorgLoop() {
 	for {
 		// Launch next background reorg if needed
 		if curDone == nil && launchNextRun {
+			// kill any currently running runReorg and launch the next one
+			if pool.interrupt != nil {
+				close(pool.interrupt)
+				pool.interrupt = nil
```

Comment:
The way we use this global channel var makes me itch. I'd prefer something like this: 97e97fc9